lib/optimizely.rb

```diff
@@ -74,13 +79,15 @@ def initialize( # rubocop:disable Metrics/ParameterLists
       config_manager = nil,
       notification_center = nil,
       event_processor = nil,
-      default_decide_options = []
+      default_decide_options = [],
+      settings = nil
     )
       @logger = logger || NoOpLogger.new
       @error_handler = error_handler || NoOpErrorHandler.new
       @event_dispatcher = event_dispatcher || EventDispatcher.new(logger: @logger, error_handler: @error_handler)
       @user_profile_service = user_profile_service
       @default_decide_options = []
+      @sdk_settings = settings
```

Comment:
Why is this declaration needed? could just be nil.
And then used in line 108.

Have you set sdk_settings to setting because shadowing the file name "sdk_settings"?

Reply:
clarified on a call, not an issue